OneSignalSDK/onesignal/src/main/java/com/onesignal/OSInAppMessageController.java

```diff
@@ -691,6 +699,17 @@ public void messageTriggerConditionChanged() {
         evaluateInAppMessages();
     }
 
+    /**
+     * If this method is called a system condition has changed to success
+     *   - Keyboard is down
+     *   - No DialogFragment visible
+     *   - Activity is on focus, this mean no prompt permissions visible
+     */
+    @Override
+    public void systemConditionChanged() {
+        checkConditionsAndDisplayFirstIAMQueued();
```

Comment:
Rename to `attemptToQueueAndShowInAppMessages()`

What if an IAM is already showing and this fires?

Reply:
inside checkConditionsAndDisplayFirstIAMQueued we check for IAM showing

Reply:
Very true

Reply:
regarding changing the name to attemptToQueueAndShowInAppMessages, in that method the message was already queue, maybe attemptToShowInAppMessage?